js/contest.js

```diff
@@ -75,6 +82,44 @@ $(document).ready(function() {
         window.location.href="contest.php?virtual=1";
     });
 
+    function updateTimeSelect(start_time,duration){
+        var lockbefore=Date.parse($("input[name=end_time]").val())-Date.parse($("input[name=lock_board_time]").val());
+        var end_time=new Date();
+        end_time.setTime(start_time+duration*1000);
+        if(end_time.toString()!="Invalid Date") $("input[name=end_time]").val($.format.date(end_time.toString(),"yyyy-MM-dd HH:mm:ss"));
+
+        var lock_time=new Date();
+        lock_time.setTime(end_time.getTime()-lockbefore);
+        if(lock_time.toString()!="Invalid Date") $("input[name=lock_board_time]").val($.format.date(lock_time.toString(),"yyyy-MM-dd HH:mm:ss"));
+    }
+    $("input[name=duration]").change(function() {
+        var duration=getDuration($(this));
+        var $text=$(this).parent().parent().parent().next().find("td");
```

Comment:
I have add id to the prompting text.. although i think rewrite that form without the use of table is the best way..